pkg/controller/baremetalhost/baremetalhost_controller_test.go

```diff
@@ -192,6 +192,29 @@ func waitForProvisioningState(t *testing.T, r *ReconcileBareMetalHost, host *met
 	)
 }
 
+// TestStatusAnnotation ensures that status is manually populated when status annotation is there.
+func TestStatusAnnotation(t *testing.T) {
```

Comment:
It would be good to have some logic to test the other cases, like to ensure the annotation is populated correctly after a reconcile and to ensure that the status isn't overwritten by the annotation if the status isn't empty.

Reply:
@dhellmann  I have added test cases for the logics you suggested. Please review